Code Cleanup

beetleskin opened the pull request:
Fixes #76 and #75

**Test Plan:**
- find missing violations of http://wiki.ros.org/CppStyleGuide (might be easier with a clang-tidy config we could also use instead of my manual refactoring? maybe as a next step ..)
- confirm no regression in functionality 
- confirm compilation on clean docker (we need a CI .. ;))

schra commented:
I'll look at this once we have merged your other PRs :)

schra commented:
Just as a reminder from #72, could you maybe add this small change in this PR too?

> When clearing the topic input, the error will be "No map received yet" - maybe we should change that to "No NavSatFix message received yet"?

beetleskin commented:
lets do that after review/concerns fixed - just remind me ;)

beetleskin commented:
#77 is also part of this PR, I'll rebase after #77 is merged

beetleskin commented:
@schra are we using a fork of https://github.com/davetcoleman/roscpp_code_format ? what about clang-format and camelCase vs. under_score? Can one set that in the config?

schra commented on src/aerialmap_display.cpp:
```diff
@@ -592,19 +589,23 @@ void AerialMapDisplay::assembleScene()
-  if (not ref_fix_)
+  if (not ref_fix_ or not center_tile_)
   {
     ROS_FATAL_THROTTLE_NAMED(2, "rviz_satellite", "ref_fix_  not set, can't create transforms");
```
`ref_fix_  not` -> `ref_fix_ not`

beetleskin replied:
I'm not getting that :/

schra replied:
yeah my input was really badly worded. fixed it myself :)

schra commented on src/aerialmap_display.h:
```diff
@@ -143,52 +150,45 @@ protected Q_SLOTS:
   std::string tile_url_;
+  /// the zoom to use (Mercator)
   int zoom_;
+  /// the number of adjacent tile blocks
```
Hm, I think this is too vague. I would say "the number of tiles loaded in each direction around the center tile"

beetleskin replied:
ok

schra commented:
> @schra are we using a fork of https://github.com/davetcoleman/roscpp_code_format ? what about clang-format and camelCase vs. under_score? Can one set that in the config? Or is that only something clang-tidy can do? Should we add a clang-tidy config? Is there one for ROS? I only know the one from moveit (https://github.com/ros-planning/moveit/blob/master/.clang-tidy).

That's a really good point. I'll add a remark about this in #78 And yeah, I think you can only do that with clang-tidy :)

beetleskin commented on src/aerialmap_display.cpp:
```diff
@@ -329,7 +319,7 @@ void AerialMapDisplay::createTileObjects()
   {
     // generate an unique name
-    static int count = 0;
+    static size_t count = 0;
```
a count is never negative, and `size_t` is `unsigned int`, but better; its encouraged to use for any counter or size: https://en.cppreference.com/w/cpp/types/size_t

schra replied:
>  its encouraged to use for any counter or size

Well, it doesn't exactly encourage it. It only states that it's typically used for looping.

Here are two reputable resources arguing against size_t (aka an unsigned integer type). One of them actually claims the exact opposite of your statement, that is that unsigneds were a good idea for counts.

* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es106-dont-try-to-avoid-negative-values-by-using-unsigned
* https://google.github.io/styleguide/cppguide.html#Integer_Types

I think that it doesn't really matter in this case too much. So if you *really* want a size_t, I'll accept it anyway.

beetleskin replied:
uff .. as you wish

beetleskin commented on src/aerialmap_display.cpp:
```diff
@@ -406,16 +407,17 @@ void AerialMapDisplay::requestTileTextures()
   {
-    tileCache_.request({ *lastCenterTile_, blocks_ });
-    dirty_ = true;
+    ROS_DEBUG_NAMED("rviz_satellite", "Requesting new tile images from the server");
```
how often does it come when you move around? it should only be hit when the center tile is updated, and for that you have to drive quite a bit (or change some params like a maniac)

beetleskin commented on src/aerialmap_display.cpp:
```diff
@@ -577,37 +589,23 @@ void AerialMapDisplay::assembleScene()
   }
+  else
+  {
+    ROS_DEBUG_NAMED("rviz_satellite", "Finished assembling all tiles");
```
I think the whole algo is not trivial due to its asyc nature. If anything goes wrong, the order counts. And as for the other debug log: this should not hit too often. Lastly, DEBUG logging is disabled by default.

beetleskin commented on src/aerialmap_display.cpp:
```diff
@@ -285,7 +286,7 @@ void AerialMapDisplay::updateTopic()
+  center_tile_ = boost::none;
   destroyTileObjects();
 
   setStatus(StatusProperty::Warn, "Message", "No map received yet");
```
done

beetleskin commented on src/mercator.h:
```diff
@@ -43,3 +43,21 @@ inline int zoomToMaxTiles(int zoom)
+ */
+inline double getTileWH(double const latitude, int const zoom) {
+  // this constant origins from how the base resolution is calculated;
+  // TODO: actually this not needed and could be removed from both formulas, since they cancel out each other;
```
check, although the same URL is present in the callee func

beetleskin commented:
> You missed two variables ;)

inferior humans .. 

I rebased, and fixed concerns - the new commit should be merged into the previous one; but I'd keep the others as they are

schra commented:
LGTM now, thanks again :)
